### ISSUE TITLE
Error in `add_html_caption()`: ! unused argument (xfun::grep_sub("^[^<]*<[^>]+aria-labelledby[ ]*=[ ]*\"([^\"]+)\".*$", "\\1", x))

### ISSUE THREAD

**Issue by marioem** [quarto-dev/quarto-cli#5762]
### Bug description

Rendering document which includes `#| tbl-cap` and `datatable` still results in error, even after #5704 .
Tested with most recent dev versions of quarto-cli and knitr.

### Steps to reproduce

````qmd
---
title: "Untitled"
subtitle: "PA2"
date: 2023-05-31
author: Me
toc: true
toc-location: left
toc-depth: 6
number-sections: true
number-depth: 6
format:
  html:
    page-layout: article
    margin-top: 2cm
    margin-bottom: 2cm
    html-math-method: katex
    code-tools: false
    self-contained: true
    smooth-scroll: true
editor: visual
execute:
  echo: false
  warning: false
  error: false
  fig-height: 6
  fig-width: 8
  fig-align: center
  fig-pos: "h"
---

## Quarto

Quarto enables you to weave together content and executable code into a finished document. To learn more about Quarto see <https://quarto.org>.

## Running Code

When you click the **Render** button a document will be generated that includes both content and the output of embedded code. You can embed code like this:

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.pos = 'h', echo=FALSE, warning=FALSE, message=FALSE, error=FALSE, fig.width=8, fig.height=6, fig.align='center')
options(xtable.comment = FALSE)
# options(knitr.table.format = "html")
options(kableExtra.latex.load_packages = FALSE)
```

```{r}
library(DT)
```

You can add options to executable code like this

```{r table}
#| column: screen-inset-right
#| tbl-cap: "Table caption"

mtcars %>% 
    datatable(extensions = 'Buttons', 
              filter = 'top',
              rownames= FALSE,
              options = list(dom = 'Blfrtip',
                             buttons = c('copy', 'csv', 'excel', 'pdf'),
                             lengthMenu = list(c(10,25,50,-1),
                                               c(10,25,50,"All"))))
```

The `echo: false` option disables the printing of code (only output is displayed).
````

### Expected behavior

Successful rendering

### Actual behavior

```
Quitting from lines 53-64 [table] (ReprEx.qmd)
Error in `add_html_caption()`:
! unused argument (xfun::grep_sub("^[^<]*<[^>]+aria-labelledby[ ]*=[ ]*\"([^\"]+)\".*$", "\\1", x))
Backtrace:
  1. global .main()
  2. execute(...)
  3. rmarkdown::render(...)
  4. knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
  5. knitr:::process_file(text, output)
     ...
 14. sew(res, options)
 15. knitr:::sew.list(x, options, ...)
 16. base::lapply(x, sew, options, ...)
 17. FUN(X[[i]], ...)
 18. knitr:::sew.knit_asis(x, options, ...)
Execution halted
```

### Quarto check output

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.2: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/emamusi/Documents/quarto-cli/package/dist/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.15 (Conda)
      Path: /Users/emamusi/miniforge3/bin/python
      Jupyter: 5.1.3
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.43.1
      rmarkdown: 2.21

[✓] Checking Knitr engine render......OK
```

**Comment by mcanouil** [comment]
Thanks for the report.

Could you update your version of Quarto, I believe you are using an old "development" version.
I can't reproduce your issue with the latest.

Also, i am pretty convinced, it's a duplicate of:
- https://github.com/quarto-dev/quarto-cli/issues/5717
- https://github.com/quarto-dev/quarto-cli/issues/5702

**Comment by mcanouil** [comment]
Because you are using the development version, yes, pull and rebuild Quarto.

**Comment by marioem** [comment]
when I issue `git pull` I'm getting 

Already up to date.

**Comment by mcanouil** [comment]
You have to rebuild Quarto as described in the readme file.

**Comment by mcanouil** [comment]
Note, that I also installed the development version of `knitr` since you are using it.
<img width="908" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/185e2d00-d428-46eb-b94b-608ae42f3030">

**Comment by marioem** [comment]
Might be DT version difference?

sessionInfo()
R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Stockholm
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0    timeDate_4022.108   dplyr_1.1.2         fastmap_1.1.1       reshape_0.8.9       GGally_2.1.2        digest_0.6.31       rpart_4.1.19        timechange_0.2.0    lifecycle_1.0.3     yardstick_1.2.0     survival_3.5-5      magrittr_2.0.3      kernlab_0.9-32     
[15] compiler_4.3.0      rlang_1.1.1         tools_4.3.0         igraph_1.4.3        utf8_1.2.3          data.table_1.14.8   knitr_1.43.1        rARPACK_0.11-0      htmlwidgets_1.6.2   xgboost_1.7.5.1     plyr_1.8.8          DiceDesign_1.9      RColorBrewer_1.1-3  parsnip_1.1.0      
[29] BiocParallel_1.34.1 withr_2.5.0         purrr_1.0.1         nnet_7.3-19         grid_4.3.0          fansi_1.0.4         colorspace_2.1-0    future_1.32.0       ggplot2_3.4.2       globals_0.16.2      scales_1.2.1        MASS_7.3-60         cli_3.6.1           ellipse_0.4.5      
[43] generics_0.1.3      rstudioapi_0.14     future.apply_1.11.0 RSpectra_0.16-1     reshape2_1.4.4      stringr_1.5.0       splines_4.3.0       dials_1.2.0         parallel_4.3.0      matrixStats_0.63.0  vctrs_0.6.2         hardhat_1.3.0       plsmod_1.0.0        Matrix_1.5-4.1     
[57] jsonlite_1.8.4      ggrepel_0.9.3       listenv_0.9.0       gower_1.0.1         tidyr_1.3.0         recipes_1.0.6       glue_1.6.2          parallelly_1.36.0   codetools_0.2-19    mixOmics_6.24.0     DT_0.28             lubridate_1.9.2     stringi_1.7.12      gtable_0.3.3       
[71] munsell_0.5.0       tibble_3.2.1        pillar_1.9.0        htmltools_0.5.5     ipred_0.9-14        lava_1.7.2.1        R6_2.5.1            lattice_0.21-8      corpcor_1.6.10      class_7.3-22        Rcpp_1.0.10         gridExtra_2.3       prodlim_2023.03.31  xfun_0.39          
[85] pkgconfig_2.0.3

**Comment by marioem** [comment]
OK, that worked, thanks. Is there any recommended way of keeping up with Quarto updates? Any feed one can subscribe to to get notifications of new releases?

**Comment by cderv** [comment]
> Is there any recommended way of keeping up with Quarto updates? Any feed one can subscribe to to get notifications of new releases?

You can watch a github repo only for its releases and get notification: https://github.blog/changelog/2018-11-27-watch-releases/
